octoprint_octobullet/__init__.py

```diff
@@ -29,10 +33,106 @@ def __init__(self):
+		self._pc_interval = 5
+		self._message_count = 0
+		self._quiet_time_sec = 1800
+		self._last_message = 0
+		self._etl_format   = "{days:d}:{hours:02d}h {minutes:02d}m"
+		self._eta_strftime = "%H:%M %d-%m"
 
 	def _connect_bullet(self, apikey, channel_name=""):
 		self._bullet, self._sender = self._create_sender(apikey, channel_name)
 
+	#~~ progress message helpers
+	def _sanitize_current_data(self, currentData):
+		if (currentData["progress"]["printTimeLeft"] == None):
+			currentData["progress"]["printTimeLeft"] = currentData["job"]["estimatedPrintTime"]
+		if (currentData["progress"]["printTimeLeft"] == None):
+			self._logger.info("Messasge: Still got no print time {}".format(currentData["progress"]["printTimeLeft"]))
+			currentData["progress"]["printTimeLeft"] = 1000
+		if (currentData["progress"]["filepos"] == None):
+			currentData["progress"]["filepos"] = 0
+		if (currentData["progress"]["printTime"] == None):
+			currentData["progress"]["printTime"] = currentData["job"]["estimatedPrintTime"]
+
+		currentData["progress"]["printTimeLeftString"] = "No ETL yet"
+		currentData["progress"]["printTimeString"] = "Not started yet"
+		currentData["progress"]["ETA"] = "No ETA yet"
+		#Add additional data
+		try:
+			currentData["progress"]["printTimeLeftString"] = self._get_time_from_seconds(currentData["progress"]["printTimeLeft"])
+			currentData["progress"]["printTimeString"] = self._get_time_from_seconds(currentData["progress"]["printTime"])
+			currentData["progress"]["ETA"] = time.strftime(self._eta_strftime, time.localtime(time.time() + currentData["progress"]["printTimeLeft"]))
+		except Exception as e:
+			self._logger.warning("Caught an exception trying to parse data: {0}\n Error is: {1}\nTraceback:{2}".format(currentData,e,traceback.format_exc()))
+			
+		return currentData
+
+	def _get_time_from_seconds(self, seconds):
+		days =0
+		hours = 0
+		minutes = 0
+		if seconds >= 86400:
+			days = int(seconds / 86400)
+			secionds = seconds % 86400
+		if seconds >= 3600:
+			hours = int(seconds / 3600)
+			seconds = seconds % 3600
+		if seconds >= 60:
+			minutes = int(seconds / 60)
+			seconds = seconds % 60
+		return self._etl_format.format(**locals())
+	
+	#~~ PrintProgressPlugin
+	def on_print_progress(self, storage, path, progress):
+		if(progress%self._pc_interval==0):
+			self._quiet_time_sec = int(self._settings.get(["quiet_minutes"])) * 60
+			if self._quiet_time_sec == 0:
+				self._quiet_time_sec = 7* 24 * 3600 # use a week if we don't want messages
+			self._logger.info("Progress: {} {} {}".format(storage,path,progress))
+			try:
+				currentData = self._printer.get_current_data()
+				currentData = self._sanitize_current_data(currentData)
+			except Exception as e:
+				self._logger.info("Caught an exception {0}\nTraceback:{1}".format(e,traceback.format_exc()))
+
+			self._logger.info("Messasge: Remaining {} ({})".format(currentData["progress"]["printTimeLeftString"],currentData["progress"]["printTimeLeftOrigin"]))
```

Comment:
All these logger lines should be `debug`, not `info`. The plugin will otherwise spam the log for every single print percentage with at least these lines, plus a bunch more further down below. That is not necessary and will make `octoprint.log` fairly unusable for debugging/post mortem analysis when this plugin is installed.

Use the log levels :) 

Also: "Messasge" instead of "Message"